src/Telemetry/Microsoft.DotNet.AzureDevOpsTimelineTests/AzureDevOpsTimelineTests.cs

```diff
@@ -0,0 +1,308 @@
+
+            /// Test execution
+            await testData.Controller.RunProject("public", 1000, CancellationToken.None);
+
+            // Test results
+            telemetryRepository.TimelineBuilds.Should().HaveCount(1);
+            telemetryRepository.TimelineBuilds[0].Build.Should().BeSameAs(builds.Single().Key);
+            telemetryRepository.TimelineBuilds[0].TargetBranch.Should().Be("theTargetBranch");
+
+            telemetryRepository.TimelineIssues.Should().BeEmpty();
+
+            telemetryRepository.TimelineRecords.Should().HaveCount(2);
+            telemetryRepository.TimelineRecords
+                .Select(r => r.Raw)
+                .Should().Contain(builds.Values
+                    .SelectMany(list => list
+                        .SelectMany(timeline => timeline.Records)));
+
+            telemetryRepository.TimelineValidationMessages.Should().BeEmpty();
+        }
+
+        /// <summary>
+        /// Build has multiple timelines. 
+        /// </summary>
+        /// <returns></returns>
+        [Test]
+        public async Task GetAdditionalTimelines()
+        {
+            // Test input
+            Dictionary<Build, List<Timeline>> builds = new Dictionary<Build, List<Timeline>>()
+            {
+                {
+                    new Build()
+                    {
+                        Id = 1,
+                        Project = new TeamProjectReference() { Name = "public" },
+                        ValidationResults = Array.Empty<BuildRequestValidationResult>(),
+                        Reason = "pullRequest",
+                        Parameters = "{\"system.pullRequest.targetBranch\": \"theTargetBranch\"}"
+                    },
+                    new List<Timeline>() {
+                        new Timeline()
+                        {
+                            Id = "1",
+                            LastChangedOn = DateTimeOffset.Now,
+                            Records = new[]
+                            {
+                                new TimelineRecord()
+                                {
+                                    Id = "1",
+                                    Issues = Array.Empty<TimelineIssue>()
+                                },
+                                new TimelineRecord()
+                                {
+                                    Id = "2",
+                                    Issues = Array.Empty<TimelineIssue>(),
+                                    PreviousAttempts = new[] {
+                                        new TimelineAttempt() { TimelineId = "2" }
+                                    }
+                                },
+                            }
+                        },
+                        new Timeline()
+                        {
+                            Id = "2",
+                            LastChangedOn = DateTimeOffset.Now.AddHours(-1),
```

Comment:
SHouldn't be using "Now" in test. That's the whole reason the SystemClock is injectable.